model.py

```diff
@@ -10,19 +10,31 @@
 from lstms import SplitLSTM
 
 class SubModel(nn.Module):
-    def __init__(self, num_const, num_var, num_hidden, num_out, num_groups):
+    def __init__(self, num_const, num_var, num_hidden, num_out, num_groups, num_batch, independent):
         super(SubModel, self).__init__()
-        self.lstm = SplitLSTM(num_const, num_var, num_hidden, num_groups, independent=True) # TODO make independent default
-        self.fc = SplitLinear(0, num_hidden, num_out, num_groups) # TODO add independent parameter for consistency
+        self.num_hidden = num_hidden
+        self.num_groups = num_groups
+        self.num_batch = num_batch
+
+        self.lstm = SplitLSTM(num_const, num_var, num_hidden, num_groups, independent)
+        self.bn = nn.BatchNorm1d(num_groups)
```

Comment:
The batch size is always equal to 1. Batch Normalization is then not effective as the running statistics are based on one sample. Batch Normalization is only worthwhile with a sufficiently large batch size. There are modified versions of BN that do work with RNNs: https://arxiv.org/pdf/1603.09025.pdf with code: https://github.com/jihunchoi/recurrent-batch-normalization-pytorch/blob/master/bnlstm.py
Maybe take a look at those (I just searched for the paper and the code for 5 minutes), for a normalization layer that works better with LSTMs specifically.

Reply:
Thanks, I'll look into it some more, I think I actually found that code too when working on it.

Reply:
It seems it is not really straightforward to use BN with LSTMs. We would have to reparameterize the LSTM like has been done for the `LSTMCell` in the code you linked. I honestly have no clue how to do that and I'm not sure if it is worth the time for us to figure it out.

Should I take out the `BatchNorm1d` for now at least as it is not really useful as you said with a batch size of 1?

Reply:
OK. Then that will be a task for next week after we've gone through the pipeline once.